### ISSUE TITLE
Uncaught TypeError: e.text.localeCompare is not a function

### ISSUE THREAD

**Issue by EminDedic95** [atom/atom#18543]
[Enter steps to reproduce:]

1. ...
2. ...

**Atom**: 1.33.0 ia32
**Electron**: 2.0.11
**OS**: Microsoft Windows 10 Education
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: e.text.localeCompare is not a function

```
At C:\Users\EminDedić\AppData\Local\atom\app-1.33.0\resources\app\static\<embedded>:11

TypeError: e.text.localeCompare is not a function
    at i (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:658095)
    at Array.sort (native)
    at SnippetsProvider.findSuggestionsForPrefix (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:657956)
    at SnippetsProvider.getSuggestions (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:657550)
    at n.forEach.t (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:647382)
    at Array.forEach (<anonymous>)
    at AutocompleteManager.getSuggestionsFromProviders (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:646898)
    at AutocompleteManager.findSuggestions (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:646636)
    at AutocompleteManager.requestNewSuggestions (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:655166)
    at AutocompleteManager.showOrHideSuggestionListForBufferChanges (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:656257)
    at Function.simpleDispatch (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:1190583)
    at Emitter.emit (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:1192024)
    at TextBuffer.emitDidChangeTextEvent (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:503956)
    at TextBuffer.transact (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:492131)
    at TextEditor.transact (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:24215)
    at mergeIntersectingSelections (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:16150)
    at TextEditor.mergeSelections (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:39484)
    at TextEditor.mergeIntersectingSelections (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:39038)
    at TextEditor.mutateSelectedText (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:16113)
    at TextEditor.insertText (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:15662)
    at TextEditor.e.(anonymous function) [as insertText] (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:259206)
    at TextEditorComponent.didTextInput (~/AppData/Local/atom/app-1.33.0/resources/app/static/<embedded>:11:93878)
```

### Commands

```
     -4:02.2.0 tree-view:open-selected-entry (div.list-tree.has-collapsable-children.file-tree-scroller.nuclide-scrollbar-style-fix.nuclide-file-tree.focusable-panel.tree-view)
  2x -3:56.9.0 script:run (input.hidden-input)
     -1:53.7.0 core:select-up (input.hidden-input)
  7x -1:42.4.0 core:backspace (input.hidden-input)
     -0:25.5.0 core:move-right (input.hidden-input)
     -0:23.5.0 editor:newline (input.hidden-input)
  2x -0:13.5.0 core:backspace (input.hidden-input)
     -0:12 autocomplete-plus:confirm (input.hidden-input)
  2x -0:10.8.0 core:backspace (input.hidden-input)
```

### Non-Core Packages

```
ask-stack 2.2.0 
atom-beautify undefined 
atom-import-js 0.21.0 
atom-js-standard 1.0.2 
atom-live-server 2.3.0 
atom-ternjs 0.18.3 
emmet 2.4.3 
file-icons 2.1.26 
gpp-compiler 3.0.7 
Hydrogen 2.6.0 
ide-python 1.0.0 
linter-gcc 0.7.1 
minimap 4.29.9 
node-debugger 1.10.1 
nuclide 0.366.0 
pigments 0.40.2 
platformio-ide-debugger 1.2.6 
platformio-ide-terminal 2.8.4 
python-debugger 0.2.0 
react 0.18.0 
react-native-snippets 0.3.0 
script 3.18.1 
```

**Comment by rsese** [comment]
Thanks for the report!  Can you confirm a few things for us?

* What were you doing at the time of the error?
* Is the error is reproducible?
* If reproducible, do you see the error in safe mode (`atom --safe`)?

x-ref: https://github.com/atom/atom/issues/18503

**Comment by guih** [comment]
@rsese I had the same issue.

Steps to reproduce:
1. Open Atom
1. Select `Javascript (Rails)` as the language.
1. Type the character `c`.

It doesn't happen if:
- Open Atom in safe mode
- Select another language

Please let me know if you need further information.

**Comment by Aerijo** [comment]
>It doesn't happen if:
>
> Open Atom in safe mode

All safe mode does is disable community packages and your init script. You should work out which package you've installed is causing the issue, and raise it on it's repo. It may still be a bug in Atom, but the maintainers will have to track it down first.

**Comment by guih** [comment]
I believe the issue is caused by the package `autocomplete-snippets` https://atom.io/packages/autocomplete-snippets.

On regular mode, when I disable this package, the error stops happening.

**Comment by Aerijo** [comment]
@guih autocomplete-snippets is a consumer for a lot of packages that will not activate if it is not active. Again, if it works in safe mode then the source of the issue is a community package.

**Comment by guih** [comment]
Thanks @Aerijo , I've tried to disable all my community packages one by one and also all together... None has fixed the issue.

Not sure if there's anywhere else to disable or if I needed to restart atom after each disable to actually disable it. Anyway, I'll keep trying. Thanks!

**Comment by Aerijo** [comment]
@guih A restart ater each disable is generally required. While packages should deactivate cleanly, many will not (nor can it be enforced) so it can't be relied upon.

I'm not sure how related this is too the original issue though; the stack trace indicates something wrong with a snippet prefix not being a string. If it is related, it may be a language package providing weird snippets?

(Good luck)

**Comment by thunderrabbit** [comment]
I triggered it by attempting to use an array where a string is needed.

In `snippets.cson` I wanted to use either 'lzz' or 'lower' to trigger this shortcut.  I was able to save snippets.cson without error, but then Atom pointed me to this page when I tried to use the snippet in a markdown file.

My `snippets.cson` file uses an array for prefix, but apparently must be a string:

```
'.source.gfm':
  'Lower Zig Zag':
    'prefix': ['lzz', 'lower']
    'body': '[Lower Zig Zag](/p/lzz)'

```
